lib/harvest.js

```diff
@@ -363,8 +364,9 @@ async function appendCollectedFilesToTaskResultsContainer(task, collection) {
   const result = await query(selectRequest);
 
   if (result.results.bindings.length) {
-    const results = await Promise.all(result.results.bindings.map(async result => {
-      const insertRequest = `
+    result.results.bindings.forEach(async result => {
+      INSERT_DATA_QUEUE.addJob(async () => {
```

Comment:
Wouldn't we have the same result, without introducing the queue, by just inserting this sequentially? (`for let of`) I am afraid now, we introduce a lot of queues, which will undermine  the initial idea of the queue, throttling the processing, so stuff don't get overloaded. (Because now, everything is going to run a bit at once)